libbeat/autodiscover/providers/jolokia/discovery.go

```diff
@@ -277,6 +277,9 @@ func (d *Discovery) update(config InterfaceConfig, message common.MapStr) {
 		return
 	}
 
+	// set instance id
+	message["id"] = agentID
+
```

Comment:
The event published for autodiscover is created in `BusEvent()`, should this id field be set there? Something like this:
```
func (e *Event) BusEvent() bus.Event {
        event := bus.Event{
                e.Type:    true,
                "id": e.AgentID,
                "jolokia": e.Message,
                "meta": common.MapStr{
                        "jolokia": e.Message,
                },
        }
        return event
}
```
And `AgentID` is added as a field to `Event`, and it could be set directly when created in `d.events <- Event{"start", agentID, message}`

Reply:
Added 2a5a1b128894fcedd5caf71422489d7e1a104111